src/routes/manage_job/get.rs

```diff
@@ -0,0 +1,180 @@
+use sqlx::types::Json;
+
+#[derive(sqlx::Type, Debug, Serialize, PartialEq, Eq)]
+#[sqlx(type_name = "valid_status", rename_all = "lowercase")]
+pub enum ValidStatus {
+	Running,
+	Completed,
+	Stopped,
+}
+
+/// Job record stores the information about a submitted job
+///
+/// `job_status` field is an update on read field. It's
+/// status will be derived from counting number of
+/// completed email verification tasks. It will be updated
+/// with the most recent status of the job.
+#[derive(sqlx::FromRow, Debug, Serialize)]
+pub struct JobRecord {
+	id: i32,
+	created_at: DateTime<Utc>,
+	total_records: i32,
+	job_status: ValidStatus,
+}
+
+/// Email record stores the result of a completed email verification task
+///
+/// It related to it's parent job through job_id
+/// It stores the result of a verification as jsonb field
+/// serialized from `CheckEmailOutput`
+#[derive(sqlx::FromRow, Debug)]
+pub struct EmailRecord {
+	job_id: i32,
+	email_id: String,
+	result: Json<CheckEmailOutput>,
+}
+
+/// Summary of a bulk verification job status
+#[derive(Debug, Serialize)]
+pub struct JobStatusSummaryResponseBody {
+	total_safe: i32,
+	total_risky: i32,
+	total_invalid: i32,
+	total_unknown: i32,
+}
+
+/// Complete information about a bulk verification job
+#[derive(Debug, Serialize)]
+pub struct JobStatusResponseBody {
+	job_id: i32,
+	created_at: DateTime<Utc>,
+	total_records: i32,
+	total_processed: i32,
+	summary: JobStatusSummaryResponseBody,
+	job_status: ValidStatus,
+}
+
+async fn job_status(
+	job_id: i32,
+	conn_pool: Pool<Postgres>,
+) -> Result<impl warp::Reply, warp::Rejection> {
+	let mut job_rec = sqlx::query_as!(
+		JobRecord,
+		r#"
+		SELECT id, created_at, total_records, job_status as "job_status: _" FROM blk_vrfy_job
+		WHERE id = $1
+		LIMIT 1
+		"#,
+		job_id
+	)
+	.fetch_one(&conn_pool)
+	.await
+	.map_err(|e| {
+		log::error!(
+			target:"reacher/v0/bulk/",
+			"Failed to get job record for [job_id={}] with [error={}]",
+			job_id,
+			e
+		);
+		ReacherError::from(e)
+	})?;
+
+	let agg_info = sqlx::query!(
+		r#"
+		SELECT
+			COUNT(*) as total_processed,
+			COUNT(CASE WHEN result ->> 'is_reachable' LIKE 'safe' THEN 1 END) as safe_count,
+			COUNT(CASE WHEN result ->> 'is_reachable' LIKE 'risky' THEN 1 END) as risky_count,
+			COUNT(CASE WHEN result ->> 'is_reachable' LIKE 'invalid' THEN 1 END) as invalid_count,
+			COUNT(CASE WHEN result ->> 'is_reachable' LIKE 'unknown' THEN 1 END) as unknown_count
+		FROM ema_vrfy_rec
+		WHERE job_id = $1
+		"#,
+		job_id
+	)
+	.fetch_one(&conn_pool)
+	.await
+	.map_err(|e| {
+		log::error!(
+			target:"reacher/v0/bulk/",
+			"Failed to get aggregate info for [job_id={}] with [error={}]",
+			job_id,
+			e
+		);
+		ReacherError::from(e)
+	})?;
+
+	if job_rec.job_status == ValidStatus::Running {
+		if job_rec.total_records == (agg_info.total_processed.unwrap() as i32) {
+			// update job status to completed
+			sqlx::query_as!(
+				JobRecord,
+				r#"
+				UPDATE blk_vrfy_job
```

Comment:
I'm not sure I like doing an update on a `GET` request. What do you think of https://github.com/reacherhq/backend/pull/254#discussion_r791017751?

Reply:
Yeah the update was only to keep the record's state consistent with the job. If there's no `job_status` field this is not needed. I'll remove it.